crates/ruff_linter/src/rules/pylint/rules/non_augmented_assignment.rs

```diff
@@ -120,36 +125,98 @@ pub(crate) fn non_augmented_assignment(checker: &mut Checker, assign: &ast::Stmt
         && ComparableExpr::from(target) == ComparableExpr::from(&value.right)
     {
         let mut diagnostic = Diagnostic::new(NonAugmentedAssignment { operator }, assign.range());
+        let half_expr = locator.slice(TextRange::new(value.left.start(), value.right.start()));
+        let right_operand_expr = trim_right_operator(half_expr);
+
         diagnostic.set_fix(Fix::unsafe_edit(augmented_assignment(
-            checker.generator(),
+            checker.locator(),
             target,
-            value.op,
-            &value.left,
+            operator,
+            right_operand_expr,
             assign.range(),
         )));
         checker.diagnostics.push(diagnostic);
     }
 }
 
+const OPERATORS: [&str; 15] = [
```

Comment:
This seems complicated and I'm not sure I follow why it's necessary. 

Could we change the logic to instead take the range from after the operator to the end of the statement instead of trying to trim the range? 

If so, I suggest using `SimpleTokenizer` to find the operator over implementing your own `Cursor` logic

Reply:
Yeah, we have a `BackwardsTokenizer` but it's not entirely clear why we even need to trim the right side. Can't we just use the statement's end?

Reply:
Oh I see. The name's are a bit misleading (or you mixed up the examples) considering that `trim_right_operator` trims the left operand? 

Either way, I think we can once use the range from after `=` up to right before the operator and everything after the operator to the statement's end